python/python/glide/async_commands/redis_modules/search/field.py

```diff
@@ -0,0 +1,385 @@
+from enum import Enum
+from typing import Dict, List, Optional
+
+
+class SortableOption(Enum):
+    """
+    Options for fields with SORTABLE argument.
+    """
+
+    SORTABLE = "SORTABLE"
+    """
+    Sets the field as sortable.
+    """
+    SORTABLE_UNF = "SORTABLE UNF"
+    """
+    Sets the field as sortable, disables the normalization and keep the original form of the field value.
+    """
+
+
+class PhoneticType(Enum):
+    """
+    Activate phonetic matching for text searches, specifies the phonetic algorithm and language used.
+    """
+
+    DM_ENGLISH = "dm:en"
+    """
+    Double metaphone for English.
+    """
+    DM_FRENCH = "dm:fn"
+    """
+    Double metaphone for French.
+    """
+    DM_PORTUGUESE = "dm:pt"
+    """
+    Double metaphone for Portuguese
+    """
+    DM_SPANISH = "dm:es"
+    """
+    Double metaphone for Spanish.
+    """
+
+
+class CoordinateSystem(Enum):
+    """
+    Specifies the coordinate system used for geo fields.
+    """
+
+    SPHERICAL = "SPHERICAL"
+    """
+    Spherical coordinates system used with latitude/longitude.
+    """
+    FLAT = "FLAT"
+    """
+    Flat coordinates system used with XY coordinates.
+    """
+
+
+class VectorAlgorithm(Enum):
+    """
+    Specifies the algorithm used for vector similarity search.
+    """
+
+    FLAT = "FLAT"
+    """
+    Brute-force index. Used for flat vector indexing.
+    """
+    HNSW = "HNSW"
+    """
+    Modified version of nmslib/hnswlib, which is an implementation of Efficient and robust 
+    approximate nearest neighbor search using Hierarchical Navigable Small World graphs.
+    """
+
+
+class Field:
+    """
+    Base class for defining fields in a schema.
+
+    Args:
+        name (str): The name of the field.
+        alias (Optional[str]): An alias for the field.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        alias: Optional[str] = None,
+    ):
+        """Initialize a new Field instance."""
+        self.name = name
+        self.alias = alias
+
+    def get_field_args(self) -> List[str]:
+        """
+        Get the arguments representing the field.
+
+        Returns:
+            List[str]: A list of field arguments.
+        """
+        args = [self.name]
+        if self.alias:
+            args.extend(["AS", self.alias])
+        return args
+
+
+class TextField(Field):
+    """
+    Class for defining text fields in a schema.
+
+    Args:
+        name (str): The name of the text field.
+        alias (Optional[str]): An alias for the field.
+        sortable (Optional[SortableOption]): If not None, sets the field as sortable with the specified sortable option. Options are: SORTABLE and SORTABLE_UNF.
```

Comment:
remove the ". Options are: SORTABLE and SORTABLE_UNF.", we don't wont to maintain another point in code if the sortable options will be changed